### PR TITLE
feat: Ledger opened app

### PR DESCRIPTION
**Pull request by maxwellmattryan**
# Description of change

Exposes new API endpoint for wallet.rs specifically for retrieving information about the currently opened app.

## Links to any relevant issues

None

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tested (Ledger Nano S) on:

- Ubuntu (20.04)
- MacOS (Catalina 10.15.7)

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/ledger.ts [hunk 1]
```diff
@@ -30,6 +31,32 @@ export function getLedgerDeviceStatus(onConnected = () => { }, onDisconnected =
...
+        })
+        .catch((err) => {
+            if(err.type === ErrorType.LedgerDeviceNotFound)
+                ledgerDeviceState.set(LedgerDeviceState.NotDetected)
+            else
+                console.error(err)
+        })
+}
+
+export function asyncGetLedgerOpenedApp(isSimulator: boolean) {
```
Comment (laumair) [inline packages/shared/lib/ledger.ts:47]: Best to specify the return type.

File: packages/shared/lib/typings/wallet.ts [hunk 2]
```diff
@@ -11,16 +11,6 @@ export interface StrongholdStatus {
     snapshotPath: string
 }
 
-export enum LedgerStatus {
```
Comment (laumair) [inline packages/shared/lib/typings/wallet.ts:14]: Good to remove it from here.

File: packages/shared/lib/validator.ts [hunk 3]
```diff
@@ -843,6 +845,7 @@ export default class ValidatorService {
      * @returns {ValidationResponse}
      */
     performValidation(response: MessageResponse | MarketDataValidationResponse | ChrysalisNodeDataValidationResponse | ChrysalisVariablesValidationResponse): ValidationResponse {
+        console.log(response, this.validators)
```
Comment (laumair) [inline packages/shared/lib/validator.ts:848]: Please remove if not needed.

File: packages/shared/lib/ledger.ts [hunk 4]
```diff
@@ -30,6 +31,32 @@ export function getLedgerDeviceStatus(onConnected = () => { }, onDisconnected =
...
+export function updateLedgerDeviceState() {
+    asyncGetLedgerOpenedApp(false)
+        .then((data: LedgerAppInfo) => {
+            ledgerDeviceState.set(data.name === 'IOTA' ? LedgerDeviceState.Connected : LedgerDeviceState.AppNotOpen)
+        })
+        .catch((err) => {
+            if(err.type === ErrorType.LedgerDeviceNotFound)
+                ledgerDeviceState.set(LedgerDeviceState.NotDetected)
+            else
+                console.error(err)
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/lib/ledger.ts:45]: Yes, without this statement Firefly can still read it as "Connected" or whichever state it's in before getting that error.

File: packages/shared/lib/ledger.ts (hunk 1 above)
Comment (maxwellmattryan) [reply packages/shared/lib/ledger.ts]: Definitely, thanks for catching this!

**Review by cvarley100 — APPROVED** [review]

**Pull request merged**